File: app/src/main/java/com/idealcn/lifecycle/study/AppApplication.java
package com.idealcn.lifecycle.study;

import android.app.Application;
import com.idealcn.lifecycle.study.greendao.source.DaoMaster;
import com.idealcn.lifecycle.study.greendao.source.DaoSession;

public class AppApplication extends Application {

    private DaoSession daoSession;

    private static AppApplication application;


    public static AppApplication get(){
        return application;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        application = this;
        daoSession =  DaoMaster.newDevSession(this,"wanandroid.db");
    }

    public DaoSession getDaoSession() {
        return daoSession;
    }
}

File: app/src/main/java/com/idealcn/lifecycle/study/AppApplication.kt
package com.idealcn.lifecycle.study

import android.app.Application
import android.arch.persistence.room.Room
import android.content.Context
import com.idealcn.lifecycle.study.room.AppRoomDatabase

class AppApplication : Application() {




    override fun onCreate() {
        super.onCreate()
       instance =  this
    }

   companion object {
       lateinit  var instance : AppApplication
       fun getAppContext() : Context {
           return  instance.applicationContext
       }



       fun getDatabase() : AppRoomDatabase{
           return Room.databaseBuilder(this.getAppContext(),AppRoomDatabase::class.java,"wanandroid.db").build()
       }
   }
}